pkg/codegen/go/gen_program.go

```diff
@@ -49,6 +49,10 @@ type generator struct {
 	configCreated       bool
 	externalCache       *Cache
 
+	// intermediate flag to generate TupleConExpression arguments correctly
+	isGenTupleConExprListArgs bool
```

Comment:
Suggestions welcome on better names/ alternate approaches to adding flags in the generator if needed @iwahbe

Reply:
I might change the name to `inGenTupleConExprListArgs`, but the benefit would be marginal. I would definitely change the comment to explain what the flag means, instead of why it exists. Something like this (but adjusted for accuracy):
```
	 // isGenTupleConExprListArgs indicates that a the generator is with a tupleConsExpression for an args list.
```